### ISSUE TITLE
Maven build: Missing requirement: com.archimatetool.canvas 4.5.1.qualifier requires 'osgi.bundle; org.eclipse.core.runtime 0.0.0' but it could not be found

### ISSUE THREAD

**Issue by yurivict** [archimatetool/archi#513]
I'm getting this on FreeBSD 12 for version 4.5.1:

```
[INFO] Computing target platform for MavenProject: com.archimatetool:com.archimatetool.canvas:4.5.1-SNAPSHOT @ /usr/ports/misc/archi/work/archi-release_4.5.1/com.archimatetool.canvas/.polyglot.build.properties
[INFO] Resolving dependencies of MavenProject: com.archimatetool:com.archimatetool.canvas:4.5.1-SNAPSHOT @ /usr/ports/misc/archi/work/archi-release_4.5.1/com.archimatetool.canvas/.polyglot.build.properties
[INFO] {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, osgi.arch=x86_64}
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: com.archimatetool.canvas 4.5.1.qualifier
[ERROR]   Missing requirement: com.archimatetool.canvas 4.5.1.qualifier requires 'osgi.bundle; org.eclipse.core.runtime 0.0.0' but it could not be found
[ERROR] 
[ERROR] See http://wiki.eclipse.org/Tycho/Dependency_Resolution_Troubleshooting for help.
[ERROR] Cannot resolve dependencies of MavenProject: com.archimatetool:com.archimatetool.canvas:4.5.1-SNAPSHOT @ /usr/ports/misc/archi/work/archi-release_4.5.1/com.archimatetool.canvas/.polyglot.build.properties: See log for details -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException
*** Error code 1
```

**Comment by jbsarrodie** [comment]
Hi. Sorry but we don't support FreeBSD. You have to contact the person who created the port.

**Comment by yurivict** [comment]
I am trying to create the port, and it breaks.

It looks like a Java error.

**Comment by jbsarrodie** [comment]
As we don't support (nor use Free BSD) we'll be of little help. It seems that this is related to eclipse, not Archi. I suggest you try to find how other people have created ports for other eclipse RCP applications (if any).

**Comment by Phillipus** [comment]
You know, rather than just copying and pasting a maven error log and saying "I'm getting this" why don't you actually say what it is you're trying to do? Saves my time and playing the "guessing game". OK, I'll play it...what Eclipse are using? How are you trying to build it? Does it compile? Why are you using Maven? Which version of Maven? Have you tried the Product export? Etc...

**Comment by yurivict** [comment]
> How are you trying to build it?

Trying to create the FreeBSD port in order to build it reproducibly.

> Does it compile?

No.

> Why are you using Maven?
I need to build it, vs. use pre-built jars. Maven seems to be the opnly option to build it.

> Which version of Maven?
maven-3.6.1

> Does it support the Maven Polyglot extension?
Probably not, polyglot-maven isn't yet ported to FreeBSD.

> Have you tried the Eclipse Product export rather than Maven?
No, this likely isn't compatible with the port's build process (it needs to be fully automatic).

> What version of Eclipse is it?
Eclipse isn't installed.

> Is it compatible with standard Eclipse 4.11?
n/a

**Comment by Phillipus** [comment]
OK, thanks for more information.

Polyglot should work with that version of Maven. Clearly the top level pom.xml file is working with Java 11 because we are using it with Travis CI to run the unit tests (https://travis-ci.org/archimatetool/archi).

We don't use Maven to build Archi, we use the Product Export. There is a comment in the pom.xml file to remove comment markers to make a build - are you doing this?

At what point in the Maven build does it fail?

**Comment by Phillipus** [comment]
Surely the main issue is creating the executable for FreeBSD and the SWT files? Do normal jar files work on FreeBSD? What exactly is different about a FreeBSD port?

**Comment by yurivict** [comment]
> There is a comment in the pom.xml file to remove comment markers to make a build - are you doing this?

I remove the markers from:
```
        <!-- <module>com.archimatetool.editor.product</module> -->
```
but it still fails.

> At what point in the Maven build does it fail?

Here is the full log: https://people.freebsd.org/~yuri/archi-failure.log

**Comment by yurivict** [comment]
> Do normal jar files work on FreeBSD? What exactly is different about a FreeBSD port

There are many ports that build jars which work fine.

**Comment by Phillipus** [comment]
I can't really help with this as I have no idea what causes the problem. Clearly the maven build is working on Linux.

**Comment by Phillipus** [comment]
Suggest comparing your build log with the one at Travis and figure out what the differences are. Also, check the maven build works on other platforms first, then try to figure out what is different.

Perhaps look for other example of Eclipse based builds for FreeBSD.

**Comment by Phillipus** [comment]
Update on this:

I just tried setting up the Archi build and test CI here on GitHub and got the same error. I tried adding a required version number to the required plug-ins but that didn't fix it.

The maven build works on Travis CI and locally.

For reference: https://wiki.eclipse.org/Tycho/Dependency_Resolution_Troubleshooting

**Comment by Phillipus** [comment]
It seems it's an issue with Maven 3.6.1. Maven 3.6.0 works,

**Comment by Phillipus** [comment]
See https://bugs.eclipse.org/bugs/show_bug.cgi?id=546463

**Comment by Phillipus** [comment]
I've tested with Maven 3.6.3 and it all seems to work now:

https://archive.apache.org/dist/maven/maven-3/3.6.3/binaries/

3.6.1 has a bug that leads to the "Missing requirement" error
3.6.2 fixed the bug in 3.6.1 but broke polyglot support

So, use either 3.6.0 or 3.6.3.

Unfortunately both Travis CI and GitHub CI Actions are using Maven 3.6.2. With Travis CI I am able to download and use 3.6.0 but I can't see how to do this with GitHub CI Actions.
